src/components/button/button.test.js

```diff
@@ -41,5 +41,102 @@ describe('/components/button', () => {
       const url = await page.url()
       expect(url).toBe(baseUrl + href)
     })
+    describe('debouncing', () => {
+      it('does not block button links', async () => {
+        // Links should take a user to another part of the page or url, so debouncing it wouldnt have any impact.
+        // But we can check that we're not blocking any initial clicks.
+
+        await page.goto(baseUrl + '/components/button/link/preview', { waitUntil: 'load' })
+
+        const href = await page.evaluate(() => document.body.getElementsByTagName('a')[0].getAttribute('href'))
+
+        // we need to start the waitForNavigation() before the keyboard action
+        // so we return a promise that is fulfilled when the navigation and the click is respectively fulfilled
+        // this is somewhat counter intuitive, as we humans expect to act and then wait for something
+        await Promise.all([
+          page.waitForNavigation(),
+          await page.click('a[role="button"]')
+        ])
+
+        const url = await page.url()
+        expect(url).toBe(baseUrl + href)
+      })
+      it('does not trigger on buttons with type=button', async () => {
+        await page.goto(baseUrl + '/components/button/preview', { waitUntil: 'load' })
+
+        // Our examples don't have form wrappers so we need to overwrite it.
+        await page.evaluate(() => {
+          document.querySelector('button').outerHTML = `
+            <form>
+              <button type="button">Save and continue</button>
```

Comment:
It'd be great to have a nicer way of doing this, but in the short term this does the job nicely 👍 

I wonder if we should wrap the existing button from the page rather than replace it though – there's a chance we could change the markup and break this functionality, but it wouldn't be picked up by the test because it'd be hardcoded to use the old markup.